scripts/screenshots.js

```diff
@@ -0,0 +1,140 @@
+const puppeteer = require('puppeteer')
+const path = require('path')
+const fs = require('fs')
+const sortBy = require('lodash/sortBy')
+
+const DEFAULT_PUPPETEER_VIEWPORT = {
+  width: 800,
+  height: 600
+}
+
+const emptyDirectory = directory => {
+  for (const filename of fs.readdirSync(directory)) {
+    fs.unlinkSync(path.join(directory, filename))
+  }
+}
+
+/**
+ * Screenshot a component to the screenshot directory, taking care of
+ * resizing the viewport before-hand so that the viewport fits the
+ * component.
+ */
+const screenshotComponent = async (page, { link, name }, screenshotDir) => {
+  await page.goto(link, { waitUntil: 'load', timeout: 0 })
+
+  // Getting the dimensions of the root wrapper
+  const rootBcr = await page.evaluate(() => {
+    const { width, height } = document
+      .querySelector('#rsg-root')
+      .getBoundingClientRect()
+    return { width, height }
+  })
+
+  // Fitting the viewport to the component
```

Comment:
If the root wrapper is responsive, won't we end up with screenshots that are always the same resolution, regardless of the component?

Reply:
The width will not changed, only the height changes to accomodate the component example if it is too long to be displayed in the default 800x600 viewport.